### PR TITLE
🔧(vite) setup vite-dts-plugin

### PR DESCRIPTION
**Pull request by jbpenrath**
## Purpose

`vite-dts-plugin` eases the configuration to extract and generate types during build. So we decide to use this library in place of `@rollup/plugin-typescript` and `tsc-alias`.

## Proposal

- [x] Remove `@rollup/plugin-typescript` and `tsc-alias` deps
- [x] Install `vite-dts-plugin`
- [x] Update vite.config.ts of `lib` package

### CODE REVIEW

File: examples/playground/vite.config.ts [hunk 1]
```diff
@@ -12,8 +12,11 @@ export default defineConfig({
   },
   plugins: [react()],
   resolve: {
-    alias: {
-      ':': resolve(__dirname, './src'),
-    },
+    alias: [
```
Comment (NathanVss) [inline examples/playground/vite.config.ts:15]: curious: why did you change to `find` syntax ?

**Review by NathanVss — COMMENTED** [review]
LGTM :)

File: examples/playground/vite.config.ts (hunk 1 above)
Comment (jbpenrath) [reply examples/playground/vite.config.ts:15]: In fact it is a debug residue. I add some issues during build and find an issue related to this setting and I tried the verbose syntax, but both syntax are currently working. Do you prefer I rollback this change ?
